app/v1/hateoas/HateoasLinks.scala

```diff
@@ -16,23 +16,40 @@
 
 package v1.hateoas
 
-import config.AppConfig
+import config.{AppConfig, FeatureSwitches}
 import v1.models.domain.{BusinessId, Nino, TaxYear}
 import v1.models.hateoas.Link
 import v1.models.hateoas.Method._
 import v1.models.hateoas.RelType._
 
 trait HateoasLinks {
 
+  private def withTaxYearParameter(appConfig: AppConfig, uri: String, maybeTaxYear: Option[TaxYear]): String = {
+    implicit val featureSwitches = FeatureSwitches(appConfig.featureSwitches)
```

Comment:
To remove warning add type annotation on `implicit`s:
```suggestion
    implicit val featureSwitches: FeatureSwitches = FeatureSwitches(appConfig.featureSwitches)
```